plugins/modules/df.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cloudera.cloud.plugins.module_utils.cdp_common import CdpModule
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: df
+short_description: Enable or Disable CDP DataFlow Services
+description:
+    - Enable or Disable CDP DataFlow Services
+author:
+  - "Dan Chaffelson (@chaffelson)"
+requirements:
+  - cdpy
+options:
+  name:
+    description: The name of the Dataflow Service
+    type: str
+    required: True
+    aliases:
+      - crn
```

Comment:
I would prefer an explicit `crn` rather than an alias, then use a `mutually_exclusive` spec.

Reply:
Both the crn and the name of the Environment are supported here, but I have made the preferred entry `crn` to be more explicit for the user. The only required submission item is the name/crn, so I don't understand why mutually_exclusive might be needed?